File: demo/java-demo/src/main/java/com/alibaba/testable/demo/DemoPrivateAccess.java
package com.alibaba.testable.demo;

import java.util.List;

/**
 * 演示私有成员访问功能
 * Demonstrate private member access functionality
 */
public class DemoPrivateAccess {

    /**
     * a private static field
     */
    private static int staticCount;
    /**
     * a private member field
     */
    private int count;
    /**
     * a constant field
     */
    public final Double pi = 3.14;

    /**
     * private static method
     */
    private static String privateStaticFunc(String str, int i) {
        return str + " + " + i;
    }

    /**
     * private member method
     */
    private String privateFunc(List<String> list, String str, int i) {
        return list.stream().reduce((a, s) -> a + s).orElse("") + " + " + str + " + " + i;
    }

}

File: demo/kotlin-demo/src/main/kotlin/com/alibaba/testable/demo/DemoPrivateAccess.kt
package com.alibaba.testable.demo

class DemoPrivateAccess {

    /**
     * a private member field
     */
    private var count = 0

    /**
     * a constant field
     */
    val pi = 3.14

    /**
     * private member method
     */
    private fun privateFunc(s: String, i: Int): String {
        return "$s - $i"
    }

    companion object {

        /**
         * a private static field
         */
        private var staticCount = 0

        /**
         * private static method
         */
        private fun privateStaticFunc(s: String, i: Int): String {
            return "$s + $i"
        }

        /**
         * private jvm static method
         */
        @JvmStatic private fun privateJvmStaticFunc(s: String, i: Int): String {
            return "$s * $i"
        }
    }

}
